cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java

```diff
@@ -72,6 +73,25 @@ public MongoBackendImpl(String mongoHosts, String mongoUsername, String mongoPas
         this.mongoHosts = mongoHosts;
         this.mongoUsername = mongoUsername;
         this.mongoPassword = mongoPassword;
+        this.mongoAuthSource = "";
+        this.dataModel = dataModel;
+    } // MongoBackendImpl
+
+    /**
+     * Constructor.
+     * @param mongoHosts
+     * @param mongoUsername
+     * @param mongoPassword
+     * @param mongoAuthSource
+     * @param dataModel
+     */
+    public MongoBackendImpl(String mongoHosts, String mongoUsername, String mongoPassword,
```

Comment:
There is no need to overload the costructor in this case, since mongoAuthSource has a not null validation on getDatabase.

Reply:
From my point of view it is better to have this new constructor to avoid side effect and maintain the backward compatibility. There is no problem to have 2 constructors in OOP and it is common to do it allowing to initialize and/or construct objects with different parameters depending of the use cases and future needs and maintaining the interface.

Reply:
I agree, but the constructor is actually only used in NGSIMongoBaseSink and in tests, so there is no need to have both.

Reply:
I agree with @IvanHdzC . The constructor is only used in one point of the code, so lest's adapt constructor and calling code better than adding a second constructor. In addition, the lesser constructor we have, the easier it would be to have good code coverage in tests.